firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -14,16 +14,33 @@
 
 package com.google.firebase.sessions
 
+import android.app.Application
+import android.util.Log
 import androidx.annotation.Discouraged
 import com.google.firebase.FirebaseApp
 import com.google.firebase.ktx.Firebase
 import com.google.firebase.ktx.app
 
-class FirebaseSessions internal constructor() {
+class FirebaseSessions internal constructor(firebaseApp: FirebaseApp) {
+  init {
+    val sessionInitiator = SessionInitiator(System::currentTimeMillis, this::initiateSessionStart)
+    val context = firebaseApp.applicationContext.applicationContext
+    if (context is Application) {
+      context.registerActivityLifecycleCallbacks(sessionInitiator.activityLifecycleCallbacks)
```

Comment:
Suggestion: Make this part of the constructor for SessionInitiator instead of exposing the `activityLifecycleCallbacks` property

Reply:
I did it this way so the SessionInitiator outer class wouldn't have any Context or Activity in the ctor to make the tests simpler.